File: composite/java/Dish.java
package composite.java;

public class Dish {
    Dish parentDish;

    void setParentDish(Dish parentDish) {
        this.parentDish = parentDish;
    }

    Dish getParentDish() {
        return parentDish;
    }

    void addDish(Dish dish) {
        // Default implementation (can be overridden by composite dishes)
        throw new UnsupportedOperationException("Cannot add dish to a leaf dish.");
    }

    void removeDish(Dish dish) {
        // Default implementation (can be overridden by composite dishes)
        throw new UnsupportedOperationException("Cannot remove dish from a leaf dish.");
    }
    boolean isComposite() {
        return false;
    }

    String prepare() {
        return "Preparing a simple dish.";
    }
}

File: composite/golang/dish.go
package main

type Dish struct {
	parentDish *Dish
}

func (d *Dish) SetParent(parent *Dish) {
	d.parentDish = parent
}

func (d *Dish) GetParent() *Dish {
	return d.parentDish
}

func (d *Dish) AddDish(dish *Dish) {
	// Default implementation: do nothing
}

func (d *Dish) RemoveDish(dish *Dish) {
	// Default implementation: do nothing
}

func (d *Dish) IsComposite() bool {
	return false
}

func (d *Dish) Prepare() string {
	return "Preparing a simple dish."
}
